blockchain/block.go

```diff
@@ -73,11 +73,11 @@ func DecodeBlock(r io.Reader) *Block {
 
 // ContainsTransaction returns true and the transaction itself if the Block
 // contains the transaction.
-func (b *Block) ContainsTransaction(t *Transaction) (bool, *Transaction) {
-	for _, tr := range b.Transactions {
+func (b *Block) ContainsTransaction(t *Transaction) (bool, uint32) {
```

Comment:
Good question. `uint32` for blocks because thats baked into the data structure, avoid overflow if we have billions of blocks. Thought we might as well keep the same datatype for transactions as well, no such thing as a negative transaction index.